File: top_interview_questions_easy/plus_one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
      for(int i = digits.size() - 1; i >= 0; i--) {
        if(digits[i] < 9) {
          // 9미만일 경우엔 그냥 1을 더한다. 
          digits[i]++;
          return digits;
        }else {
          // 그렇지 않을 경우엔 0을 넣는다. 
          digits[i] = 0;
        }
      }
      // 9초과로 인해 여기로 온 값은 가장 뒤 값이 0으로 초기화가 된다. 
      digits.push_back(0);
      // 그리고 맨 앞의 값은 1이 된다. 
      digits[0] = 1;
      return digits;
    }
};

File: top_interview_questions_easy/plus_one.py
  """
  # approach
  마지막 index 리스트를 찾는다.
  해당 인덱스에 1을 더하고 해당 요소가 10이 넘을 경우 올림 수를 저장하고
  바로 앞 요소에 더한다.

  # answer
  리스트를 문자로 바꾼 다음 join()을 이용해 문자열로 합쳐준다.
  합쳐준 문자열을 숫자로 바꾼 다음 +1
  그런 후 문자열로 바꾸어 리스트로 만들어 반환
  """
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        digits = list(map(str, digits))
        nums = int(''.join(digits))
        nums += 1
        nums = list(str(nums))
        return list(map(int, nums))

